packages/perspective-jupyterlab/src/ts/renderer.ts

```diff
@@ -128,8 +151,20 @@ export class PerspectiveDocumentWidget extends DocumentWidget {
                         this.context.model.fromString(resultAsB64);
                         this.context.save();
                     } else if (this._type === "json") {
+                        // TODO: Gotta fix this - need to do something about Date values
+
                         const result = await view.to_json();
-                        this.context.model.fromJSON(result);
+                        for (let [_key, val] of Object.entries(result)) {
+                            if (val instanceof Date) {
+                                throw new Error("TODO: Dates unsupported");
```

Comment:
this needs to be addressed. As of right now I don't recall why I added this loop to begin with - some cockamamie scheme in anger to satisfy the type checker? (this loop would be narrowing some of the types involved)

Reply:
Perspective outputs `number` values in posix epoch time for columns of type `date` and `datetime`, ergo I think the fix for this is to remove `Date` from the [types of these methods in `@finos/perspective`](https://github.com/finos/perspective/blob/master/packages/perspective/index.d.ts#L106).